### ISSUE TITLE
Game crashes when forging Iron Anvil 

### ISSUE THREAD

**Issue by dakotaseeker** [anegostudios/VintageStory-Issues#659]
**Game Version:** 1.14.0-rc.3
**Platform:** Windows /
**Modded:** No 

### Description
Game crashing when forging the Iron Anvil Top, seems to be triggered by trying to move specific voxels. Reproducing the same movements upon reloading would cause a crash on the same voxel as the previous crash, but making different moves will result in a different voxel becoming the trigger eventually.

### How to reproduce
Continue smithing an iron anvil top.

### Expected behavior
If not obvious, what did you expect to happen?

### Screenshots
Attach a screenshot if it helps explain the issue / shows a visual bug.

### Logs
```
Running on 64 bit Windows with 0 GB RAM 
Version: v1.14.0-rc.3 (Unstable)11/24/2020 1:19:20 PM: Critical error occurred
System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at Vintagestory.GameContent.BlockEntityAnvil.OnUpset(Vec3i voxelPos, BlockFacing towardsFace)
   at Vintagestory.GameContent.BlockEntityAnvil.OnUseOver(IPlayer byPlayer, Vec3i voxelPos, BlockSelection blockSel)
   at Vintagestory.GameContent.ItemHammer.OnHeldAttackStop(Single secondsPassed, ItemSlot slot, EntityAgent byEntity, BlockSelection blockSel, EntitySelection entitySel)
   at Vintagestory.Client.NoObf.SystemMouseInWorldInteractions.HandleHandInteraction(Single dt)
   at Vintagestory.Common.EventManager.TriggerGameTick(Int64 ellapsedMilliseconds, IWorldAccessor world)
   at Vintagestory.Client.NoObf.ClientMain.MainRenderLoop(Single dt)
   at Vintagestory.Client.NoObf.ClientMain.MainGameLoop(Single deltaTime)
   at _AlHq0Inapm61pWcZ59khYnLNZ4o._JVIQA9L1AmYKQRSbCtldwFFuYBu(Single )
   at _3oBqZxtXFzGwmJ7quAnuEUCIsWB._l5fK2kGCHgpDIf3cGe4QLBObbKC(Single )
   at _3oBqZxtXFzGwmJ7quAnuEUCIsWB._oTv4ofbpa4AGXLcoYrtTL6GbHQf(Single )
   at Vintagestory.Client.NoObf.ClientPlatformWindows.window_RenderFrame(Object sender, FrameEventArgs e)
   at System.EventHandler`1.Invoke(Object sender, TEventArgs e)
   at OpenTK.GameWindow.RaiseRenderFrame(Double elapsed, Double& timestamp)
   at OpenTK.GameWindow.DispatchRenderFrame()
   at OpenTK.GameWindow.Run(Double updates_per_second, Double frames_per_second)
   at _4BWchexiFigD9Q16Uzu1AHRcMkKA._F6I3lsgrc5VRUFEGnBoXsboBGFs(_zaqE2EDRPzgKzO45PKPCgTGhJk6 , String[] )
   at _ZkFtsNKa9prG34Jd8zC0yw7G6T4._F6I3lsgrc5VRUFEGnBoXsboBGFs(ThreadStart )
-------------------------------
[client-crash.txt](https://github.com/anegostudios/VintageStory-Issues/files/5591911/client-crash.txt)

In the game's Logs folder you may be able to find more information
inside client-main.txt and server-main.txt. Alternatively, you may use
gist.github.com to upload logs, linking them here, if they're too large.

Note that the crash log contains all crashes that have happened so far.
You likely only want to report the latest one.
```

**Comment by tyronx** [comment]
thanks for the report, can you show me a screenshot of that anvil and which voxel causes the crash?

**Comment by dakotaseeker** [comment]
Unfortunately i don't have one, but the same voxel thing would only happen
if i repeated the exact same pattern after reloading. If i moved the voxels
around in a different way then i would end up crashing on a different voxel.

**Comment by tyronx** [comment]
hm, it would save me a lot of time if i had an easy reproducible case.

**Comment by dakotaseeker** [comment]
sorry, i didn't think to save anything from this. However i do still have the crash logs, it happened a bunch of times, would that help?
[client-crash.txt](https://github.com/anegostudios/VintageStory-Issues/files/5625280/client-crash.txt)

**Comment by Craluminum2413** [comment]
I guess it's fixed in 1.15.0-pre.3
> Fixed: BlockFacing.FromNormal(vec3i) returned wrong values in some cases

**Comment by tyronx** [comment]
That fix is not related, but I would assume this to be fixed anyway, since we have no new reports in half a year
